gdnative-core/src/lib.rs

```diff
@@ -58,6 +58,7 @@ pub mod init;
 pub mod log;
 pub mod object;
 pub mod profiler;
+pub mod register;
```

Comment:
Just to be clear, is this intended to replace what we currently have in `mod export` completely, instead of just an extension? The two words are very similar in meaning, so this naming is a possible source of confusion if the modules are intended to coexist.

Reply:
Yes. Separate to avoid major breaking change.

Reply:
By "yes", do you mean that this is intended to replace what we currently have in `mod export` completely? If so, then I'm afraid it would be hard to get this merged as is. Please see my reply down below.

Reply:
Sorry, I didn't read the intent of your question.

Eventually replace, but keep both for some time to prevent breaking change and provide time for migration.